### ISSUE TITLE
Aggregating peaks doesn't work at all

### ISSUE THREAD

**Issue by maxim-h** [PMBio/muon#39]
**Describe the bug**

While going through the tutorial on the 10xMultiome data and plotting the PCA with values for cut counts in peaks corresponding to different genes all plots are colored the same.

![pic-selected-210820-1614-40](https://user-images.githubusercontent.com/22867431/130249406-4c51e779-13fd-4108-9d35-3cf4285b0852.png)

**Expected behaviour**
Correct aggregation across specific peak sets.

**System**
 - OS: `Linux 5.13.12-arch1-1 #1 SMP PREEMPT Wed, 18 Aug 2021 20:49:03 +0000 x86_64 GNU/Linux`
 - Python version [3.9]
 
 <details>
<summary>  Versions of libraries involved</summary>

alabaster==0.7.12
anndata==0.7.6
argon2-cffi==20.1.0
async-generator==1.10
attrs==21.2.0
Babel==2.9.1
backcall==0.2.0
bleach==3.3.1
certifi==2021.5.30
cffi==1.14.6
charset-normalizer==2.0.4
click==8.0.1
cycler==0.10.0
debugpy==1.4.1
decorator==4.4.2
defusedxml==0.7.1
docutils==0.16
entrypoints==0.3
flit==3.3.0
flit_core==3.3.0
h5py==3.3.0
idna==3.2
imagesize==1.2.0
ipykernel==6.0.3
ipython==7.25.0
ipython-genutils==0.2.0
ipywidgets==7.6.3
jedi==0.18.0
Jinja2==3.0.1
joblib==1.0.1
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==6.1.12
jupyter-console==6.4.0
jupyter-core==4.7.1
jupyterlab-pygments==0.1.2
jupyterlab-widgets==1.0.0
jupyterthemes==0.20.0
kiwisolver==1.3.1
leidenalg==0.8.7
lesscpy==0.15.0
llvmlite==0.36.0
loompy==3.0.6
MarkupSafe==2.0.1
matplotlib==3.4.2
matplotlib-inline==0.1.2
mistune==0.8.4
muon @ file:///[my local path]/muon
natsort==7.1.1
nbclient==0.5.3
nbconvert==6.1.0
nbformat==5.1.3
nbsphinx==0.8.7
nest-asyncio==1.5.1
networkx==2.5.1
notebook==6.4.0
numba==0.53.1
numexpr==2.7.3
numpy==1.21.1
numpy-groupies==0.9.13
packaging==21.0
pandas==1.3.1
pandocfilters==1.4.3
parso==0.8.2
patsy==0.5.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==8.3.1
ply==3.11
prometheus-client==0.11.0
prompt-toolkit==3.0.19
protobuf==3.17.3
ptyprocess==0.7.0
pycparser==2.20
Pygments==2.9.0
pynndescent==0.5.4
pyparsing==2.4.7
pyrsistent==0.18.0
pysam==0.16.0.1
python-dateutil==2.8.2
python-igraph==0.9.6
pytz==2021.1
pyzmq==22.1.0
qtconsole==5.1.1
QtPy==1.9.0
readthedocs-sphinx-search==0.1.0
requests==2.26.0
scanpy==1.8.1
scikit-learn==0.24.2
scipy==1.7.0
seaborn==0.11.1
Send2Trash==1.7.1
sinfo==0.3.4
six==1.16.0
sklearn==0.0
snowballstemmer==2.1.0
Sphinx==4.1.2
sphinx-automodapi==0.13
sphinx-rtd-theme==0.5.2
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
statsmodels==0.12.2
stdlib-list==0.8.0
tables==3.6.1
terminado==0.10.1
testpath==0.5.0
texttable==1.6.4
threadpoolctl==2.2.0
toml==0.10.2
tornado==6.1
tqdm==4.61.2
traitlets==5.0.5
umap-learn==0.5.1
urllib3==1.26.6
wcwidth==0.2.5
webencodings==0.5.1
widgetsnbextension==3.5.1
xlrd==1.2.0

</details>

**Additional context**

I believe it's related to #21 or it's fix. In fact reverting the b2347af produces correct behavior. 

![pic-selected-210820-1605-17](https://user-images.githubusercontent.com/22867431/130249762-a7da5e2e-1f66-4185-aec3-cc5b820ed915.png)

I've checked manually and the subsetting used in b2347af doesn't work, i.e. returns full matrix (top of the picture) with AnnData 0.7.6.

![pic-selected-210820-1604-24](https://user-images.githubusercontent.com/22867431/130250079-c2ce8d9a-1418-4aa5-b14e-9b2442008289.png)

**Comment by gtca** [comment]
Thanks for reporting that, @maxim-h. Indeed, `adata[:, peaks].raw` does not perform the desired subsetting, `.raw` slots have their own features.

@ilia-kats is it something that we can still fix for backed objects on our end? At the very least I believe we should at least fix it for non-backed objects in the meantime.
